components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.ts

```diff
@@ -376,16 +375,27 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
     this.expressionForm.addControl('identityProvider', new FormControl('', Validators.required));
   }
 
-  private addNameControl(): void {
-    this.expressionForm.addControl('name', new FormControl('',
-      [
-        Validators.required,
-        Validators.pattern(Regex.patterns.NO_MIXED_WILDCARD)
-      ]
-    )
-    );
+  private addNameControl(identityProvider): void {
+    if (identityProvider === 'ldap' || identityProvider === 'saml' ) {
+        this.expressionForm.addControl('name', new FormControl('',
+          [
+            Validators.required,
+            Validators.pattern(Regex.patterns.NO_MIXED_WILDCARD_ALLOW_SPECIAL)
+          ]
+        )
+      );
+    } else {
+        this.expressionForm.addControl('name', new FormControl('',
+          [
+            Validators.required,
+            Validators.pattern(Regex.patterns.NO_MIXED_WILDCARD_ALLOW_HYPHEN)
+          ]
+        )
+      );
+    }
```

Comment:
Updated the addNameControl logic to accept the identityProvider.  All we are really doing here is checking to see if it is ldap/saml or not.

I could probably simplify this by interpolating the pattern type... think i'll try that right now actually.

Reply:
scratch that, I think this is better left untouched